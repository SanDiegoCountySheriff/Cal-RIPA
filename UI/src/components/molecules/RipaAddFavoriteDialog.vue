<template>
  <v-dialog v-model="model" max-width="500px" persistent>
    <v-card>
      <v-card-title>
        <span>Add Favorite</span>
      </v-card-title>

      <v-card-text>
        <v-container>
          <v-row>
            <v-col cols="12">
              <v-text-field
                v-model="favoriteName"
                label="Favorite Name"
                required
              ></v-text-field>
            </v-col>
          </v-row>
        </v-container>
      </v-card-text>

      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="blue darken-1" text @click="handleClose"> Cancel </v-btn>
        <v-btn
          color="blue darken-1"
          :disabled="getSaveDisabled"
          text
          @click="handleSave"
        >
          Save
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: 'ripa-add-favorite-dialog',

  data() {
    return {
      viewModel: this.showDialog,
      favoriteName: '',
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
      set(newValue) {
        if (!newValue) {
          this.onClose()
        }
        this.viewModel = newValue
      },
    },
    getSaveDisabled() {
      return this.favoriteName.length === 0
    },
  },

  methods: {
    init() {
      this.favoriteName = ''
    },

    handleClose() {
      this.onClose()
    },

    handleSave() {
      this.onAddFavorite(this.favoriteName)
      this.handleClose()
    },
  },

  created() {
    this.init()
  },

  watch: {
    showDialog(newValue) {
      this.viewModel = newValue
    },
  },

  props: {
    showDialog: {
      type: Boolean,
      default: false,
    },
    onAddFavorite: {
      type: Function,
      required: true,
    },
    onClose: {
      type: Function,
      required: true,
    },
  },
}
</script>
