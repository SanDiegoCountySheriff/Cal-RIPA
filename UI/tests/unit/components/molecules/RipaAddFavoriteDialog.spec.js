import RipaAddFavoriteDialog from '@/components/molecules/RipaAddFavoriteDialog'
import { shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa Add Favorite Dialog', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = propsData => {
    return shallowMount(RipaAddFavoriteDialog, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaAddFavoriteDialog, {
      vuetify,
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set model', async () => {
    wrapper = factory()
    wrapper.vm.model = true
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.viewModel).toEqual(true)
    expect(wrapper.emitted('on-close')).toBeFalsy()

    wrapper.vm.model = null

    expect(wrapper.emitted('on-close').length).toEqual(1)
  })

  it('should get saved disabled', () => {
    wrapper = factory()

    expect(wrapper.vm.getSaveDisabled).toBeTruthy()
    wrapper.vm.favoriteName = 'Test'

    expect(wrapper.vm.getSaveDisabled).toBeFalsy()
  })

  it('should handle close', () => {
    wrapper = factory()

    wrapper.vm.handleClose()

    expect(wrapper.emitted('on-close').length).toEqual(1)
  })

  it('should handle save', () => {
    wrapper = factory()

    wrapper.vm.handleSave()

    expect(wrapper.emitted('on-add-favorite').length).toEqual(1)
    expect(wrapper.emitted('on-close').length).toEqual(1)
  })

  it('should watch show dialog', async () => {
    wrapper = factory()

    wrapper.setProps({ showDialog: true })
    await wrapper.vm.$nextTick()

    expect(wrapper.vm.viewModel).toBeTruthy()
  })
})
