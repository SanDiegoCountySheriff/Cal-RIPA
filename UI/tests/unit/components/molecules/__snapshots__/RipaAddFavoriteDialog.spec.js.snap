// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Add Favorite Dialog should match snapshot 1`] = `
<div class="v-dialog__container">
  <!---->
</div>
`;
